### PR TITLE
Add option to search for unlisted packages

### PR DESCRIPTION
**Pull request by twsouthwick**
This change exposes the options for searching in a more centralized way so that we can add more if we need to without breaking the API.

### CODE REVIEW

**Comment by twsouthwick** [comment]
@sunandabalu The builds don't seem to be showing up as checks for my PR. Any ideas?

File: src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/TargetCompatibilityReferenceAnalyzer.cs [hunk 1]
```diff
@@ -51,7 +51,7 @@ public async Task AnalyzeAsync(IProject project, IDependencyAnalysisState state,
                 else
                 {
                     // If the package won't work on the target Framework, check newer versions of the package
-                    var newerVersions = await _packageLoader.GetNewerVersionsAsync(packageReference, state.TargetFrameworks, true, token).ConfigureAwait(false);
+                    var newerVersions = await _packageLoader.GetNewerVersionsAsync(packageReference, state.TargetFrameworks, new() { Prerelease = true }, token).ConfigureAwait(false);
```
Comment (mjrousos) [inline src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/TargetCompatibilityReferenceAnalyzer.cs:54]: I think the `true` that was passed here previously corresponded to `latestMinorAndBuildOnly`, not `prerelease`.

**Review by mjrousos — CHANGES_REQUESTED** [review]
Good change. Only concern is that I've think you've set the wrong option property in TargetCompatibilityReferenceAnalyzer.

File: src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/TargetCompatibilityReferenceAnalyzer.cs (hunk 1 above)
Comment (sunandabalu) [reply src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/TargetCompatibilityReferenceAnalyzer.cs:54]: Mike is right, it was for latestMinorAndBuildOnly

File: src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/PackageMapReferenceAnalyzer.cs [hunk 2]
```diff
@@ -99,14 +99,14 @@ private bool ContainsPackageReference(IEnumerable<NuGetReference> packages, stri
...
-        private async Task AddNetCoreReferences(NuGetPackageMap packageMap, IDependencyAnalysisState state, IProject project, CancellationToken token)
+        private async Task AddNetCoreReferences(NuGetPackageMap packageMap, IDependencyAnalysisState state, CancellationToken token)
         {
             foreach (var newPackage in packageMap.NetCorePackages)
             {
                 var packageToAdd = newPackage;
                 if (packageToAdd.HasWildcardVersion)
                 {
-                    var reference = await _packageLoader.GetLatestVersionAsync(packageToAdd.Name, state.TargetFrameworks, false, token).ConfigureAwait(false);
+                    var reference = await _packageLoader.GetLatestVersionAsync(packageToAdd.Name, state.TargetFrameworks, new(), token).ConfigureAwait(false);
```
Comment (sunandabalu) [inline src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/PackageMapReferenceAnalyzer.cs:109]: no need to set Prerelease false to the options object here? will it default to false?

**Comment by sunandabalu** [comment]
> @sunandabalu The builds don't seem to be showing up as checks for my PR. Any ideas?

Looks like there is an AzDO outage happening, they are working on mitigating it.

File: src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/PackageMapReferenceAnalyzer.cs (hunk 2 above)
Comment (twsouthwick) [reply src/steps/Microsoft.DotNet.UpgradeAssistant.Steps.Packages/Analyzers/PackageMapReferenceAnalyzer.cs:109]: yes, that's the default for bool values

**Review by mjrousos — APPROVED** [review]
LGTM

**Pull request merged**